src/hooks/useSearchParams.tsx

```diff
@@ -0,0 +1,25 @@
+import { useRouter } from 'next/router';
+import { useMemo } from 'react';
+
+export const useSearchParams = () => {
+  const { query, pathname } = useRouter();
+
+  return useMemo(() => {
+    const params = new URLSearchParams();
+
+    const filtered = Object.entries(query).filter(([key]) => {
+      return !pathname.includes(`[${key}]`);
+    });
+
+    filtered.forEach(([key, value]) => {
+      if (Array.isArray(value)) {
+        value.forEach((v) => params.append(key, v));
+        return;
+      }
+
+      params.set(key, value);
+    });
+
+    return params;
+  }, [query, pathname]);
+};
```

Comment:
I thought this was [in Next already](https://nextjs.org/docs/13/app/api-reference/functions/use-search-params) or is it doing an extra bit of logic maybe? 🤔

Reply:
Only from nextjs 13 😄 we're on 12.x. Once we update this we'll be able to remove our hook